src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -1092,3 +1118,43 @@ function getSessionId(): string {
 
     return sessionId;
 }
+
+async function createClientPipeTransport(
+    pipeName: string,
+    encoding: RAL.MessageBufferEncoding = 'utf-8'
+): Promise<PipeTransport> {
+    const connected = new Promise<[MessageReader, MessageWriter]>((resolve) => {
+        const server: net.Server = net.createServer((socket: net.Socket) => {
+            // Sever has connected
+            server.close();
+            resolve([new SocketMessageReader(socket, encoding), new SocketMessageWriter(socket, encoding)]);
+        });
+
+        // Start the server listening on the pipe name.
+        const pipeConnectionString = getClientPipeName(pipeName);
+        server.listen(pipeConnectionString);
+    });
+
+    // Provide onConnected so that calling method can spin up the pipe client
+    return {
+        onConnected: async () => {
+            return await connected;
+        },
+    };
+}
+
+function getClientPipeName(pipeName: string): string {
+    const isWindows = os.platform() === 'win32';
+
+    if (isWindows) {
+        // Windows pipes are named using the following format
+        return '\\\\.\\pipe\\' + pipeName;
```

Comment:
Probably good to make a comment here that the pipeName doesn't include the prefix because node and dotnet have slightly different prefixes when talking about the name.

Reply:
Updated to create all names on the client, and included this comment... 

```
    // The connection string format for Nodejs and .NET are different.
    const WINDOWS_NODJS_PREFIX = '\\\\.\\pipe\\';
    const WINDOWS_DOTNET_PREFIX = '\\\\.\\';
```